utils.py

```diff
@@ -46,13 +73,21 @@ def upload_image_helper(image_data, bucket_name):
         aws_access_key_id=os.environ["SPACES_ACCESS_KEY_ID"],
         aws_secret_access_key=os.environ["SPACES_SECRET_ACCESS_KEY"],
     )
-
-    res = client.put_object(
-        Bucket=bucket_name,
-        Key=img_filename,
-        Body=BytesIO(img_data),
-        ACL="public-read",
-    )
+    
+    if file_data:
+        client.put_object(
+            Bucket=bucket_name,
+            Key=img_filename,
+            Body=file_data.read(),
+            ACL="public-read",
+        )
+    else:
+        client.put_object(
+            Bucket=bucket_name,
+            Key=img_filename,
+            Body=BytesIO(img_bytes),
+            ACL="public-read",
+        )
```

Comment:
Also essentially duplicated code that can be factored out. I think a more intuitive way to structure this code would be to create a helper upload method and then simple methods for base64 and file upload that just pass necessary params into the helper

Would also make the function spec easier to understand